### ISSUE TITLE
Unable to use with `tokio::spawn()` ("the trait `std::marker::Send` is not implemented")

### ISSUE THREAD

**Issue by aisrael** [Empty2k12/influxdb-rust#11]
Am trying to use `influxdb` within a Hyper/tokio request handler. Since your InfluxDB client is asynchronous, I figured it'd be a better choice (to keep everything async).

However, when I try to run the following code in my hyper service:

```
    let client = InfluxDbClient::new("http://localhost:8086", "test");

    let write_query =
        InfluxDbQuery::write_query(Timestamp::NOW, "weather").add_field("temperature", 82);

    tokio::spawn(client.query(&write_query));
```

I promptly get:

```
error[E0277]: `dyn futures::future::Future<Error = influxdb::error::InfluxDbError, Item = std::string::String>` cannot be sent between threads safely
  --> src/arboric/mod.rs:59:5
   |
59 |     tokio::spawn(client.query(&write_query));
   |     ^^^^^^^^^^^^ `dyn futures::future::Future<Error = influxdb::error::InfluxDbError, Item = std::string::String>` cannot be sent between threads safely
   |
   = help: the trait `std::marker::Send` is not implemented for `dyn futures::future::Future<Error = influxdb::error::InfluxDbError, Item = std::string::String>`
   = note: required because of the requirements on the impl of `std::marker::Send` for `std::ptr::Unique<dyn futures::future::Future<Error = influxdb::error::InfluxDbError, Item = std::string::String>>`
   = note: required because it appears within the type `std::boxed::Box<dyn futures::future::Future<Error = influxdb::error::InfluxDbError, Item = std::string::String>>`
   = note: required by `tokio::executor::spawn`
```

Am unsure of how to proceed. Can I just fork `influxdb-rust` and add `+ Send` to the return type of `InfluxDbClient::query()`? Is the return value _even_ `Send`-able, or should I clone it?

(Pardon me if this should be something obvious, am quite a beginner still with Rust + tokio + futures.)

**Comment by Empty2k12** [comment]
Sorry for my late response, I somehow didn't get a notification from Github.

I just tried adding `+ Send` and it still compiled. I would appreciate if you tested with adding `+ Send` and seeing if your code compiles and works as expected. You can then open a PR or I can add this.

**Comment by Empty2k12** [comment]
@aisrael Is there any updates on this? Did you get it working with adding a `+ Send`?

**Comment by aisrael** [comment]
Oh hang on let me try and do a quick MWE.

**Comment by aisrael** [comment]
Oh hang on let me try and do a quick MWE.

Here, this works!
https://github.com/aisrael/tokio-influxdb/blob/master/src/main.rs

**Comment by Empty2k12** [comment]
Great. `+ Send` can also be added in a lot of other places then. Would you create a PR for that or should I  do that?

**Comment by Drevoed** [comment]
Should you really return Box from the query function? You can return impl Future and use Either inside that function. Also if you're comfortable with it I'm up for converting this library to beta async-await version using alpha tokio and futures-preview crates in another branch.

**Comment by Empty2k12** [comment]
@Drevoed I was thinking about opening an issue for tokio 0.2 and reqwest async, but since it's not stable yet I didn't do that yet. I'd like to keep compatibility with stable, but I'd be up for having a beta version 👍

**Comment by Drevoed** [comment]
@Empty2k12 Ok then I'm already converting chunks of code and also making some optimizations, for example you don't need to always call for Reqwest Client::new and instead keep it in InfluxdbClient

**Comment by Empty2k12** [comment]
@Drevoed Much appreciated! I wrote this library because of a need we had and I'm by no means an expert in Rust, so I'm happy about all suggestions for improvements. Also please feel free to review pull requests, the more eyes the better!

**Comment by Empty2k12** [comment]
Fixed in #32
